src/components/hooks/useSetQueryParams.tsx

```diff
@@ -0,0 +1,20 @@
+import { useEffect, useState } from 'react';
+import useRouter from './useRouter';
+
+export default function useSetQueryParams(
+  object?: Record<string, string>
+): [Record<string, string>, (item: Record<string, string>) => void] {
+  const [searchParams, setSearchParams] = useState(object || {});
+  const { push, pathname } = useRouter();
+
+  useEffect(() => {
+    if (searchParams && Object.keys(searchParams).length) {
```

Comment:
Wciąż nie jestem przekonany do używania tutaj useEffect, ale niech zostanie. Poszukamy lepszego rozwiązania później.

Reply:
Poprawiłem to wczesniej i teraz odpala się tylko raz, dwa jesli jest parametr search czyli tak jak powinno byc